deployment/src/main/java/io/quarkus/jgit/deployment/GiteaDevServiceInfoBuildItem.java

```diff
@@ -1,38 +1,62 @@
 package io.quarkus.jgit.deployment;
 
+import java.net.URI;
+import java.net.URL;
+
 import io.quarkus.builder.item.SimpleBuildItem;
 
 /**
  * A build item that represents the information required to connect to a Gitea dev service.
  */
 public final class GiteaDevServiceInfoBuildItem extends SimpleBuildItem {
 
+    private final URL url;
     private final String host;
     private final int httpPort;
     private final String adminUsername;
     private final String adminPassword;
 
-    public GiteaDevServiceInfoBuildItem(String host, int httpPort, String adminUsername, String adminPassword) {
+    public GiteaDevServiceInfoBuildItem(String url, String adminUsername, String adminPassword) {
+        this(createURL(url), adminUsername, adminPassword);
+    }
+
+    public GiteaDevServiceInfoBuildItem(URL url, String adminUsername, String adminPassword) {
+        this(url, url.getHost(), url.getPort(), adminUsername, adminPassword);
+    }
+
+    public GiteaDevServiceInfoBuildItem(URL url, String host, int httpPort, String adminUsername, String adminPassword) {
+        this.url = url;
         this.host = host;
         this.httpPort = httpPort;
         this.adminUsername = adminUsername;
         this.adminPassword = adminPassword;
     }
 
-    public int httpPort() {
-        return httpPort;
+    private static URL createURL(String url) {
```

Comment:
You can infer the URL from the host and port, so there is no need to pass it as a parameter

Reply:
Also I avoided instantiating a new URL because it would internally parse it using `URLStreamHandler`s and that's not necessary here

Reply:
Provided, that the protocol is fixed. 😄 
In this case I think its safe to assume it's always `http`, so I'll infer.